traits/trait_types.py

```diff
@@ -3336,6 +3338,145 @@ def as_ctrait(self):
         """
         return self.trait_maker.as_ctrait()
 
+# -------------------------------------------------------------------------------
+#  'None' trait:
+# -------------------------------------------------------------------------------
+
+
+class NoneTrait(TraitType):
+    """ Defines a trait that only accepts the None value
+    """
+
+    info_text = "None"
+
+    default_value = None
+
+    default_value_type = DefaultValue.constant
+
+    def __init__(self, **metadata):
+        default_value = metadata.pop("default", None)
+        if default_value is not None:
+            raise ValueError("Cannot set default value {} "
+                             "for NoneTrait".format(default_value))
+        super(NoneTrait, self).__init__(**metadata)
+
+    def validate(self, obj, name, value):
+        if value is None:
+            return value
+
+        self.error(obj, name, value)
+
+# -------------------------------------------------------------------------------
+#  'Union' trait:
+# -------------------------------------------------------------------------------
+
+
+class Union(TraitType):
+    """ Defines a trait whose value can be any of of a specified list of
+    trait types or list of trait type instances or None
+    """
+    # CTrait type map for special trait types:
+    type_map = {"event": TraitKind.event, "constant": TraitKind.constant}
+
+    def __init__(self, *traits, **metadata):
+        super(Union, self).__init__(traits, **metadata)
```

Comment:
You don't want to pass `traits` as an argument here, I think - it will be treated as the default value.

I would suggest the following strategy for this method:

- process the traits to get the list of ctraits and set the class attribute
- work out what the default value is. If the default value comes from a ctrait (or is None) you should be able to also get the default value type.
- call `super` with default value and metadata.

Reply:
Also, for Python 3, you can use `super()` rather than `super(Union, self)`